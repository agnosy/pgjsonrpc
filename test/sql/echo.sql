\set ECHO none
BEGIN;
\t
\i sql/pgjsonrpc.sql
\set ECHO all

-- You should write your tests

SELECT jsonrpc.execute('{"id":1, "method":"test"}');

SELECT jsonrpc.execute('{"jsonrpc": "2.0", "id":1, "method":"echo", "params": {"message": "Hello jsonrpc!"}}');

INSERT INTO jsonrpc.methods(name, description, function_name)
VALUES('echotest', 'Echoes the parameter passed to echo.', 'jsonrpc.echo');

SELECT jsonrpc.execute('{"jsonrpc": "2.0", "id":1, "method":"echotest", "params": {"message": "Hello jsonrpc!"}}');

ROLLBACK;
