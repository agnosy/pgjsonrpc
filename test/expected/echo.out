\set ECHO none
-- You should write your tests
SELECT jsonrpc.execute('{"jsonrpc": "2.0", "id":1, "method":"echo", "params": {"message": "Hello jsonrpc!"}}');
 {"id" : "1", "jsonrpc" : "2.0", "result" : "Hello jsonrpc!"}

INSERT INTO jsonrpc.methods(name, description, function_name)
VALUES('echotest', 'Echoes the parameter passed to echo.', 'jsonrpc.echo');
SELECT jsonrpc.execute('{"jsonrpc": "2.0", "id":1, "method":"echotest", "params": {"message": "Hello jsonrpc!"}}');
 {"id" : "1", "jsonrpc" : "2.0", "result" : "Hello jsonrpc!"}

ROLLBACK;
